src/cmd-tag

```diff
@@ -0,0 +1,211 @@
+
+
+def main():
+    """Main entry point"""
+
+    args = parse_args()
+    args.func(args)
+
+
+def parse_args():
+    """Parse args and dispatch"""
+
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest="cmd", title="Tag actions")
+    subparsers.required = True
+
+    create = subparsers.add_parser("create", help="Create a tag from a build")
+    create.add_argument("--tag", help="Tag to be created", required=True)
+    create.add_argument("--build", help="Build to be tagged", required=True)
+    create.add_argument("--force", help="Force tag creation",
+                        action="store_true")
+    create.set_defaults(func=cmd_create)
+
+    delete = subparsers.add_parser("delete",
+                                   help="Delete a tag from the metadata")
+    delete.add_argument("--tag", help="Tag to delete from metadata",
+                        required=True)
+    delete.set_defaults(func=cmd_delete)
+
+    list_tags = subparsers.add_parser("list", help="List available tags in "
+                                                   "the metadata")
+    list_tags.set_defaults(func=cmd_list)
+
+    update = subparsers.add_parser("update", help="Update a tag to new build")
+    update.add_argument("--tag", help="Tag to be updated", required=True)
+    update.add_argument("--build", help="Build to update tag with",
+                        required=True)
+    update.add_argument("--force", help="Force the update of a tag",
+                        action="store_true")
+    update.set_defaults(func=cmd_update)
+
+    return parser.parse_args()
+
+
+def init_build_data():
+    """Initialize build metadata"""
+
+    builds_json = "builds/builds.json"
+
+    if os.path.isfile(builds_json):
+        with open(builds_json) as f:
+            builddata = json.load(f)
+    else:
+        print("INFO: Did not find existing builds.json; starting with empty "
+              "build list and tags list")
+        builddata = {"builds": [], "tags": []}
+
+    return builddata
+
+
+def finalize_json(builddata):
+    """Wrapper around cmdlib.write_json()"""
+
+    builds_json = "builds/builds.json"
+
+    builddata['timestamp'] = rfc3339_time()
+    write_json(builds_json, builddata)
+
+
+def available_builds(builddata):
+    """Returns the availble build IDs from build metadata"""
+
+    return builddata["builds"]
+
+
+def available_tags(builddata):
+    """Returns the available tag names from build metadata"""
+
+    available_tag_names = []
+    if "tags" in builddata.keys():
+        for t in builddata["tags"]:
+            available_tag_names.append(t["name"])
+
+    return available_tag_names
+
+
+def cmd_list(args):
+    """List available tags in build metadata"""
+
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+    if not avail_tags:
+        print("No tags found")
+        sys.exit(0)
+
+    for t in build_data["tags"]:
+        print(f"name: {t['name']}\n"
```

Comment:
While the output on this would be slightly different would it make sense to simplify it via something like:

```
for t in build_data["tags"]:
    print(json.dump({"name": t.get("name"), "created": t.get("created"), "target": t.get("target")}))
```

or

```
print(json.dumps({t.get("name"): {"created": t.get("created"), "target": t.get("target")} for t in build_data["tags"]})
```

Reply:
I'm going to switch it out...we'll see what the people think